File: growing-shop-java/src/main/java/com/example/growingshop/domain/auth/dto/AuthRequest.java
package com.example.growingshop.domain.auth.dto;

import com.example.growingshop.domain.company.domain.Company;
import com.example.growingshop.domain.user.domain.User;
import com.example.growingshop.domain.user.domain.UserGrade;
import com.example.growingshop.domain.user.domain.UserStatus;
import com.example.growingshop.domain.user.domain.UserType;
import com.example.growingshop.global.validator.StringChecker;
import lombok.*;

import javax.persistence.Column;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

public class AuthRequest {

    @Getter
    @NoArgsConstructor(access = AccessLevel.PROTECTED)
    @AllArgsConstructor
    @Builder
    public static class LoginReq {
        @StringChecker(includeLowerEn = true, includeUpperEn = true, includeNumber = true, includeSpecialCharacter = "-_")
        @Column(nullable = false)
        private String loginId;

        @Column(nullable = false)
        private String password;
    }

    @NoArgsConstructor(access = AccessLevel.PROTECTED)
    @AllArgsConstructor
    @Getter
    @Builder
    public static class JoinReq {
        @NotBlank
        @Size(max = 30)
        @StringChecker(includeLowerEn = true, includeUpperEn = true, includeKo = true, includeSpecialCharacter = "'-.")
        private String name;

        @Size(max = 13, min = 9)
        @NotBlank
        @Pattern(regexp = "^[0-9]{2,3}-?[0-9]{3,4}-?[0-9]{4}$")
        private String mobile;

        @Email
        @NotBlank
        private String email;

        @StringChecker(includeLowerEn = true, includeUpperEn = true, includeNumber = true, includeSpecialCharacter = "-_")
        @Column(nullable = false)
        private String loginId;

        @Column(nullable = false)
        private String password;

        private Long company;

        public User toEntity(String hashedPassword, Company company) {
            return User.builder()
                    .name(this.name)
                    .mobile(this.mobile)
                    .email(this.email)
                    .loginId(this.loginId)
                    .password(hashedPassword)
                    .company(company)
                    .status(company != null ? UserStatus.UNDER_REVIEW : UserStatus.NORMAL)
                    .type(company != null ? UserType.SELLER : UserType.NORMAL)
                    .grade(UserGrade.NORMAL)
                    .build();
        }
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/auth/dto/AuthRequest.kt
package com.example.growingshopauth.auth.dto

import com.example.domain.company.Company
import com.example.domain.user.User
import com.example.domain.user.UserGrade
import com.example.domain.user.UserStatus
import com.example.domain.user.UserType
import jakarta.persistence.Column
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

class AuthRequest {

    class LoginReq(
        @Pattern(regexp = "^[a-zA-Z가-힣-_]+$")
        @Column(nullable = false)
        val loginId: String,

        @Column(nullable = false)
        val password: String
    )

    class JoinReq(
        @NotBlank
        @Size(max = 30)
        @Pattern(regexp = "^[a-zA-Z가-힣'-.']+$")
        val name: String,

        @Size(max = 13, min = 9)
        @NotBlank
        @Pattern(regexp = "^[0-9]{2,3}-?[0-9]{3,4}-?[0-9]{4}$")
        val mobile: String,

        @Email
        @NotBlank
        val email: String,

        @Pattern(regexp = "^[a-zA-Z가-힣'-_]+$")
        @Column(nullable = false)
        val loginId: String,

        @Column(nullable = false)
        val password: String,

        val company: Long? = null
    ) {
        fun toEntity(hashedPassword: String, company: Company?): User {
            return User(
                name = this.name,
                mobile = this.mobile,
                email = this.email,
                loginId = this.loginId,
                password = hashedPassword,
                company = company,
                status = if (company != null) UserStatus.UNDER_REVIEW else UserStatus.NORMAL,
                type = if (company != null) UserType.SELLER else UserType.NORMAL,
                grade = UserGrade.NORMAL
            )
        }
    }
}
